### DISCUSSION TITLE
EMQx-tls-setup

### DISCUSSION THREAD

**Discussion by Suryaobulareddy** [emqx/emqx#6213] [Q&A] [answered]
Hi Guy,

I'm trying to setup EMQx with TLS for EMQx Broker.

I tried all the possibilities which are officially available in the website and as well as from other sources, but no luck

If any tried, please help me

Thank You

**Comment by Suryaobulareddy** [comment, marked as answer]
For you're reference attaching the logs snapshot below

![image](https://user-images.githubusercontent.com/33251273/142268748-8f2e9db2-03e0-45bd-b6c4-f8b1589aef2e.png)

**Reply by zmstone** [reply]
eacces hints emqx has no permission to read the file(s)

**Reply by Suryaobulareddy** [reply]
Thank for the reply...

I do provided 777 permissions to all the certs but no luck

Is there any folder specific, like we have to maintain the certificates only in /etc/emqx/certs

I'm using EMQ X Broker 4.3.5

**Reply by zmstone** [reply]
Maybe change the full path to owned by `emqx` user, not just the files.
There is a private key, 777 is not recommended.

EMQ X runs by emqx user, try `sudo su emqx` to test access.

**Reply by Suryaobulareddy** [reply]
Thank You @zmstone 

It is due the ownership problem..

Now it is working fine for me

I have another query. Can you please help me?

How to check whether emqx tls is 1 way ssl or 2 way ssl enable?

**Reply by zmstone** [reply]
Configure broker to `verify_peer` will make broker check client’s certificates. 
for clients to check broker’s it’s client side configuration
